src/problems/Albany_Layouts.cpp

```diff
@@ -128,42 +128,42 @@ Albany::Layouts::Layouts (int worksetSize, int numVertices, int numNodes, int nu
   isSideLayouts = true;
 
   // Collapsed sideset layouts are now used to ensure contiguous memory access for efficient GPU evaluation
-  sidesetWorksetSize = singleWorksetSizeAllocation ? sidesetWorksetSize : worksetSize*numSides;
+  // sidesetWorksetSize = singleWorksetSizeAllocation ? sidesetWorksetSize : worksetSize*numSides;
 
   // Solution Fields
-  node_scalar  = rcp(new MDALayout<Side,Node>(sidesetWorksetSize,numNodes));
-  qp_scalar    = rcp(new MDALayout<Side,QuadPoint>(sidesetWorksetSize,numQPts));
-  cell_scalar  = rcp(new MDALayout<Side,QuadPoint>(sidesetWorksetSize,1));
-  cell_scalar2 = rcp(new MDALayout<Side>(sidesetWorksetSize));
+  node_scalar  = rcp(new MDALayout<Side,Node>(sideSetName,0,numNodes));
+  qp_scalar    = rcp(new MDALayout<Side,QuadPoint>(sideSetName,0,numQPts));
+  cell_scalar  = rcp(new MDALayout<Side,QuadPoint>(sideSetName,0,1));
+  cell_scalar2 = rcp(new MDALayout<Side>(sideSetName,0));
 
-  node_vector = rcp(new MDALayout<Side,Node,Dim>(sidesetWorksetSize,numNodes,vecDim));
-  qp_vector   = rcp(new MDALayout<Side,QuadPoint,Dim>(sidesetWorksetSize,numQPts,vecDim));
-  cell_vector = rcp(new MDALayout<Side,Dim>(sidesetWorksetSize,vecDim));
+  node_vector = rcp(new MDALayout<Side,Node,Dim>(sideSetName,0,numNodes,vecDim));
+  qp_vector   = rcp(new MDALayout<Side,QuadPoint,Dim>(sideSetName,0,numQPts,vecDim));
+  cell_vector = rcp(new MDALayout<Side,Dim>(sideSetName,0,vecDim));
```

Comment:
Let's try to make these `PHX::DataLayout` like we discussed. It looks like the string name will have to be unique for each layout in that case:
https://github.com/trilinos/Trilinos/blob/master/packages/phalanx/example/FiniteElementAssembly_MixedFieldTypes/FiniteElementAssembly_MixedFieldTypes.cpp
I'm curious to see if we still get layout sizes if we dump debug info in this case.

Reply:
I switched the side set layout types to `PHX::DataLayout` as we discussed and now fields aren't showing up at all when I call `getFieldTagsForSizing`. I'm not sure why this would be the case. Do you want me to push the current commit so you can take a look at the code? I still think the change I mentioned to MDALayout on Phalanx's end would be a much more elegant solution and is likely the intended functionality.

Here is an example log. Note that this build has the print statements that print all (side set) field tags found in the field manager.
[log-ctest-dynamic-sizing.1012570.txt](https://github.com/SNLComputation/Albany/files/6934415/log-ctest-dynamic-sizing.1012570.txt)

Reply:
can you dump all field tag identifiers? yes, go ahead and push the code. I wonder if the fm is having issues with a mix of MDA and Data layouts.

Reply:
It looks like the sideset tags are there but they don't have sizes in the identifier. Not sure why they weren't getting printed before.

Log: 
[log-ctest-dynamic-sizing.1012589.txt](https://github.com/SNLComputation/Albany/files/6934775/log-ctest-dynamic-sizing.1012589.txt)

Reply:
Okay yeah I think that's correct. Is it possible that `first_dim_name` is not being set?

Could you also dump the phalanx graph to see if the sizes show up there?
`Phalanx Graph Visualization Detail: 1` in the input file